http/src/config.rs

```diff
@@ -146,10 +131,36 @@ fn create(
         },
     };
 
-    serde_json::to_writer_pretty(BufWriter::new(config), &config_contents)
+    let config_path = ipfs_path.join("config");
+
+    let config_file = fs::create_dir_all(&ipfs_path)
+        .map_err(InitializationError::DirectoryCreationFailed)
+        .and_then(|_| {
+            OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(&config_path)
+                .map_err(InitializationError::ConfigCreationFailed)
+        })?;
+
+    let mut writer = BufWriter::new(config_file);
+
+    serde_json::to_writer_pretty(&mut writer, &config_contents)
         .map_err(InitializationError::ConfigWritingFailed)?;
 
-    Ok(())
+    writer.flush().unwrap();
```

Comment:
```suggestion
    writer.flush().map_err(InitializationError::ConfigCreationFailed)?;
```

Not too critical though, but as the the errors have been used around here already, it'd be inline with other functionality. The `ConfigCreationFailed` isn't a really good match for this of course.. Another way would be to change the type of `ConfigWritingFailed` to be `Box<dyn std::error::Error + 'static>` and it'd probably work just as fine.

Reply:
Ah yes, was wondering about this, thanks 👍